pkg/proxy/iptables/proxier_test.go

```diff
@@ -2864,31 +2864,45 @@ func TestMasqueradeRule(t *testing.T) {
 	}
 }
 
-func TestOnlyLocalExternalIPs(t *testing.T) {
+// TestExternalTrafficPolicyLocal tests that traffic to externally-facing IPs does not get
+// masqueraded when using Local traffic policy. For traffic from external sources, that
+// means it can also only be routed to local endpoints, but for traffic from internal
+// sources, it gets routed to all endpoints.
+func TestExternalTrafficPolicyLocal(t *testing.T) {
 	ipt := iptablestest.NewFake()
 	fp := NewFakeProxier(ipt)
+
 	svcIP := "172.30.0.41"
 	svcPort := 80
+	svcNodePort := 3001
+	svcHealthCheckNodePort := 30000
 	svcExternalIPs := "192.168.99.11"
+	svcLBIP := "1.2.3.4"
 	svcPortName := proxy.ServicePortName{
 		NamespacedName: makeNSN("ns1", "svc1"),
 		Port:           "p80",
 	}
 
 	makeServiceMap(fp,
 		makeTestService(svcPortName.Namespace, svcPortName.Name, func(svc *v1.Service) {
```

Comment:
**TestOnlyLocalLoadBalancing** had test services with session affinity and assertIPTablesRulesEqual validated the rule. 

I could see only three tests that check session affinity translation to iptable rules 
**TestOnlyLocalLoadBalancing**, **TestTerminatingEndpointsTrafficPolicyLocal** and **TestTerminatingEndpointsTrafficPolicyCluster** and it looks like we are removing session affinity from all these tests. We might loose coverage, right ?

Reply:
nvm, the first commit covers it I guess https://github.com/kubernetes/kubernetes/pull/120829/commits/43db55e93de797a5f2b3fe3af90a424c32d80f00

Reply:
sorry, missed this last week. Yeah, I added session affinity to the "overall" test specifically to make up for the lost coverage elsewhere